libraries/AP_IOMCU/AP_IOMCU.cpp

```diff
@@ -1038,18 +1038,24 @@ void AP_IOMCU::check_iomcu_reset(void)
 
     if (dt_ms < max_delay) {
         // all OK
+        last_safety_off = reg_status.flag_safety_off;
         return;
     }
     detected_io_reset = true;
     INTERNAL_ERROR(AP_InternalError::error_t::iomcu_reset);
     hal.console->printf("IOMCU reset t=%u %u %u dt=%u\n",
                         unsigned(AP_HAL::millis()), unsigned(ts1), unsigned(reg_status.timestamp_ms), unsigned(dt_ms));
 
-    if (safety_forced_off && !reg_status.flag_safety_off && hal.util->get_soft_armed()) {
-        // IOMCU has reset while armed with safety off - force it off
-        // again so we can keep flying
-        force_safety_off();
+    if (last_safety_off && !reg_status.flag_safety_off && hal.util->get_soft_armed()) {
+        AP_BoardConfig *boardconfig = AP_BoardConfig::get_singleton();
+        uint16_t options = boardconfig?boardconfig->get_safety_button_options():0;
```

Comment:
We really should be raising internal errors in a lot of these places that find `boardconfig` is nullptr.

Won't ever happen in here AFAICS based on the code sequencing in the various `init_ardupilot`s - but if it did then we wouldn't retry this block and might not reset the safety switch appropriately.